includes/library/zencart/listingBox/src/Box/AbstractListingBox.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * Class AbstractListingBox
+ *
+ * @copyright Copyright 2003-2014 Zen Cart Development Team
+ * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
+ * @version $Id: currencies.php 15880 2010-04-11 16:24:30Z wilt $
+ */
+namespace ZenCart\ListingBox\Box;
+/**
+ * Class AbstractListingBox
+ * @package ZenCart\ListingBox\Box
+ */
+abstract class AbstractListingBox extends \base
+{
+    /**
+     * @var
+     */
+    protected $productQuery;
+    /**
+     * @var
+     */
+    protected $outputLayout;
+    /**
+     * @var
+     */
+    protected $formatter;
+    /**
+     * @var
+     */
+    protected $paginator;
+    /**
+     * @var
+     */
+    protected $templateVariables;
+
+    /**
+     *
+     */
+    public abstract function __construct();
+
+    /**
+     * @param $diContainer
+     */
+    public function init($diContainer)
```

Comment:
`init` methods are bad form.  A static factory method is better with dependencies defined in the constructor:
```php
public function __construct($formatter, QueryBuilder $db)
{
    $this->formatter = $formatter;
    $this->db = $db;
}

public static function build(Container $container)
{
    $formatter = $container->get('formatter');
    $db = $container->get('queryBuilder');
    return new static($formatter, $db);
}
```